src/function/table/arrow.cpp

```diff
@@ -304,6 +317,63 @@ void ArrowToDuckDBList(Vector &vector, ArrowArray &array, ArrowScanState &scan_s
 	ColumnArrowToDuckDB(child_vector, *array.children[0], scan_state, list_size, arrow_convert_data, col_idx,
 	                    list_col_idx, start_offset);
 }
+
+void ArrowToDuckDBBlob(Vector &vector, ArrowArray &array, ArrowScanState &scan_state, idx_t size,
+                       std::unordered_map<idx_t, unique_ptr<ArrowConvertData>> &arrow_convert_data, idx_t col_idx,
+                       idx_t &list_col_idx, idx_t nested_offset) {
+	auto original_type = arrow_convert_data[col_idx]->variable_sz_type[list_col_idx++];
+	SetValidityMask(vector, array, scan_state, size, nested_offset);
+	if (original_type.first == ArrowVariableSizeType::FIXED_SIZE) {
+		//! Have to check validity mask before setting this up
+		idx_t offset = (scan_state.chunk_offset + array.offset) * original_type.second;
+		if (nested_offset != 0) {
+			offset = original_type.second * nested_offset;
+		}
+		auto cdata = (char *)array.buffers[1];
+		for (idx_t row_idx = 0; row_idx < size; row_idx++) {
+			if (FlatVector::IsNull(vector, row_idx)) {
+				continue;
+			}
+			auto bptr = cdata + offset;
+			auto blob_len = original_type.second;
+			FlatVector::GetData<string_t>(vector)[row_idx] = StringVector::AddStringOrBlob(vector, bptr, blob_len);
+			offset += blob_len;
+		}
+	} else if (original_type.first == ArrowVariableSizeType::NORMAL) {
+		auto offsets = (uint32_t *)array.buffers[1] + array.offset + scan_state.chunk_offset;
+		if (nested_offset != 0) {
+			offsets = (uint32_t *)array.buffers[1] + array.offset + nested_offset;
+		}
+		auto cdata = (char *)array.buffers[2];
+		for (idx_t row_idx = 0; row_idx < size; row_idx++) {
+			if (FlatVector::IsNull(vector, row_idx)) {
+				continue;
+			}
+			auto bptr = cdata + offsets[row_idx];
+			auto blob_len = offsets[row_idx + 1] - offsets[row_idx];
+			FlatVector::GetData<string_t>(vector)[row_idx] = StringVector::AddStringOrBlob(vector, bptr, blob_len);
+		}
+	} else {
+		//! Check if last offset is higher than max uint32
+		if (((uint64_t *)array.buffers[1])[array.length] > std::numeric_limits<uint32_t>::max()) {
```

Comment:
Use NumericLimits<uint32_t>::Maximum() to avoid upsetting Windows.

Reply:
Yeah I keep forgetting that